site/docs/features/index.md

```diff
@@ -1,107 +1,327 @@
-.. record 1 ..
-.. record 2 ..
-.. only 2 records ..
+To get around that issue, jobs can create a new "work"-branch when they start.
+The results from all tasks of a job are recorded as individual commits into that
+"work"-branch. Once the job has finished, all changes are then merged into the 
+"master" branch at once.
+```
+    "work"
+    branch
+      |
+      |
+      v
++-----------+ 
+| commit #1 | 
++-----------+ 
+      ^
+      |
+      |
+   "master"
+    branch
+```
+Our example Spark job has two tasks, each generates a separate commit, which are only
+visible on our "work"-branch:
+```
+          task#1         task#2   "work"
+          result         result   branch
+            |                |     |
+            v                v     v
+      +-----------+       +-----------+
+      | commit #2 | --<-- | commit #3 |
+      +-----------+       +-----------+
+         |
+         v
+         |
++-----------+ 
+| commit #1 | 
++-----------+ 
+      ^
+      |
+      |
+   "master"
+    branch
+```
+When the job has finished, you can merge the now consistent result back
+into the "master"-branch.
 ```
+          task#1         task#2   "work"
+          result         result   branch
+            |                |     |
+            v                v     v
+      +-----------+       +-----------+
+      | commit #2 | --<-- | commit #3 |
+      +-----------+       +-----------+
+         |                          |        
+         v                          ^
+         |                          |
++-----------+                     +-----------+
+| commit #1 | --------<---------- | commit #4 |  
++-----------+                     +-----------+
+                                      ^
+                                      |
+                                      |
+                                   "master"
+                                    branch
+```
+Note that the merge-`commit #4` has references to two commits:
+
+* the previous commit on the "master" branch and
+* the merged commit from the "work"-branch
+
+It would be nice to give that `commit #4` a meaningful commit message, like
+`aggregate-financial-stuff 2020/12/24`, though, so people that look through the
+history of the data can grasp what that commit changes and why it's there.
+
+Before you actually merge the data from a working-branch into the "master" branch,
+it is possible to review all the changes made by your job - either programmatically,
+like running some process that validates the state of the branch, or asking a human
+to take a look.
+
+### Working branches for "humans"
+
+You can also use "developer" branches to run experiments against your data, test
+changes of your jobs, etc etc etc.
+
+Production and staging and development can use the same data lake without risking
+the consistent state of production data.
+
+### Squashing commits
+
+Nessie can also help to reduce the amount of data files in a data lake.
```

Comment:
Nessie can't (yet) squash commits, would rather not add this section until it can

Reply:
Added a sentence that squashing isn't supported yet.
Or do you prefer to completely remove the section?

Reply:
i think i would remove it for the time being, don't feel to strongly tho